### PR TITLE
Fix registering party duplicate + Refactoring components

### PR DESCRIPTION
**Pull request by orelbn**
*Issue #:* /bcgov/entity#16097

*Description of changes:*
- Added `isRegisteringParty` param to `isExistingSecuredParty`, to distinguish  between registered and secured parties
- Refactored code to change styling from module to scoped
- Refactored code to use props directly rather than first converting to localState
- Refactored code to use functional programming rather than loops
- Moved repeated checks to local state.
- Update test to utilize testing utils and added tests
- Moved composables to composables directory
- Moved util function to composable state as it does not act as a util function (not a general purpose function) `isSecuredPartyRestrictedList` => `isSecuredPartiesRestricted`

*Screenshots*:

Before:
Can not add same secured party as registering party if added secured party first:

image

When manually trying to add not exact:

image

When adding exact:

image

After:

Can add same secured and registering party:

image

When manually trying to add:

image

When adding exact:

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1423-wtw0979p.web.app

File: ppr-ui/src/components/parties/party/EditParty.vue [hunk 1]
```diff
@@ -378,13 +370,13 @@ export default defineComponent({
           currentSecuredParty.value.personName.last = ''
         }
         // check for duplicate
-        if (hasMatchingSecuredParty(currentSecuredParty.value, props.isEditMode)) {
+        if (!props.isRegisteringParty && hasMatchingSecuredParty(currentSecuredParty.value, props.isEditMode)) {
```
Comment (orelbn) [inline ppr-ui/src/components/parties/party/EditParty.vue]: Does not trigger duplicate for when adding a Registering Party, as there can only be one

File: ppr-ui/src/composables/parties/useSecuredParty.ts [hunk 2]
```diff
@@ -90,12 +93,12 @@ export const useSecuredParty = (props, context) => {
     }
   }
 
-  const isExistingSecuredParty = (partyCode: string): boolean => {
-    let parties = getAddSecuredPartiesAndDebtors.value // eslint-disable-line
-    const idx = parties.securedParties.findIndex(party =>
-      party.code === partyCode
-    )
-    return idx !== -1
+  const isExistingSecuredParty = (partyCode: string, isRegisteringParty: boolean): boolean => {
```
Comment (orelbn) [inline ppr-ui/src/composables/parties/useSecuredParty.ts:96]: Added prop to check if it isRegisteringParty

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1423-wtw0979p.web.app

**Review by cameron-eyds — APPROVED** [review]
Looks great, a lot of clean up here! 
Hopefully the test coverage was comprehensive enough to give confidence in all these refactors :) 
As always but especially in this case, flag Divya for regression testing on the modified parts 👍

**Comment by orelbn** [comment]
> Looks great, a lot of clean up here! Hopefully the test coverage was comprehensive enough to give confidence in all these refactors :) As always but especially in this case, flag Divya for regression testing on the modified parts 👍

Will do thanks for the reminder.

File: ppr-ui/src/composables/parties/useSecuredParty.ts [hunk 3]
```diff
@@ -20,7 +21,7 @@ const initAddress = {
 }
 const { isPartiesValid } = useParty()
 
-export const useSecuredParty = (props, context) => {
+export const useSecuredParty = (props?, context?) => {
```
Comment (dimak1) [inline ppr-ui/src/composables/parties/useSecuredParty.ts]: If `props` and `context` are optional, we probably should set optional chaining for them on line 63, 92?

**Review by dimak1 — APPROVED** [review]
Looks great!

File: ppr-ui/src/composables/parties/useSecuredParty.ts (hunk 3 above)
Comment (orelbn) [reply ppr-ui/src/composables/parties/useSecuredParty.ts]: I removed props, and just passed activeIndex to the functions that use it, then optionally chained context.

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1423-wtw0979p.web.app

**Pull request merged**